tests/model_serving/model_server/private_endpoint/conftest.py

```diff
@@ -0,0 +1,195 @@
+import json
+import pytest
+from typing import Generator
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.secret import Secret
+from ocp_resources.namespace import Namespace
+from ocp_resources.pod import Pod
+from simple_logger.logger import get_logger
+from ocp_resources.service_mesh_member import ServiceMeshMember
+from ocp_resources.serving_runtime import ServingRuntime
+from kubernetes.dynamic import DynamicClient
+
+from utilities.serving_runtime import ServingRuntimeFromTemplate
+from tests.model_serving.model_server.storage.pvc.utils import create_isvc
+from tests.model_serving.model_server.private_endpoint.utils import create_sidecar_pod, get_flan_pod, b64_encoded_string
+from tests.model_serving.model_server.private_endpoint.infra import create_ns
+from tests.model_serving.model_server.private_endpoint.constants import (
+    AWS_REGION_EAST_2,
+    AWS_ENDPOINT_EAST_2,
+)
+
+
+LOGGER = get_logger(name=__name__)
+
+
+@pytest.fixture(scope="module")
+def endpoint_namespace(admin_client: DynamicClient) -> Generator[Namespace, None, None]:
+    yield from create_ns(admin_client=admin_client, name="endpoint-namespace")
+
+
+@pytest.fixture(scope="module")
+def diff_namespace(admin_client: DynamicClient) -> Generator[Namespace, None, None]:
+    yield from create_ns(admin_client=admin_client, name="diff-namespace")
+
+
+@pytest.fixture()
+def endpoint_sr(
+    admin_client: DynamicClient,
+    endpoint_namespace: Namespace,
+) -> Generator[ServingRuntime, None, None]:
+    with ServingRuntimeFromTemplate(
+        client=admin_client,
+        name="flan-example-sr",
+        namespace=endpoint_namespace.name,
+        template_name="caikit-tgis-serving-template",
+    ) as model_runtime:
+        yield model_runtime
+
+
+@pytest.fixture()
+def endpoint_s3_secret(
+    admin_client: DynamicClient,
+    endpoint_namespace: Namespace,
+    aws_access_key_id: str,
+    aws_secret_access_key: str,
+    s3_bucket_name_wisdom: str,
+) -> Generator[Secret, None, None]:
+    data = {
+        "AWS_ACCESS_KEY_ID": b64_encoded_string(aws_access_key_id),
+        "AWS_DEFAULT_REGION": b64_encoded_string(AWS_REGION_EAST_2),
```

Comment:
This is the conftest for private_endpoint, so I think it makes sense to not handle every possible case. If we were to generalize this for any test then I'd agree with you that it would need to handle other regions

Reply:
we need to get the region from the user as the bucket has a region